packages/toc/src/generators/python/index.ts

```diff
@@ -63,6 +63,26 @@ function generate(editor: IDocumentWidget<FileEditor>): IHeading[] {
       });
     };
   }
+
+  /**
+   * Clean a line of code that declares either a class or function/method
+   * keeps text until first open parenthesis or first colon
+   * i.e. strips out arguments of functions and superclass specification of classes
+   *
+   * @private
+   * @param line - line of python code to clean
+   * @returns cleaned line of code
+   */
+  function clean_declaration_code_string(line: string): string {
+    let first_paren = line.indexOf('(');
+    let first_colon = line.indexOf(':');
+    let last_char = Math.min(
+      first_paren || Infinity,
+      first_colon || Infinity,
+      line.length
+    );
+    return line.slice(0, last_char);
```

Comment:
This is not working as expected. `indexOf` returns -1 if the character is not found. So `-1 || Infinity` is _-1_. And therefore `last_char` = -1.

```suggestion
    const candidates = [line.indexOf('('), line.indexOf(':'), line.length];
    return line.slice(0, Math.min(...candidates.filter(n => n >= 0)));
```